### DISCUSSION TITLE
can table be dropped successfully when database engine is atomic and has continually insert

### DISCUSSION THREAD

**Discussion by congbaoyangrou** [ClickHouse/ClickHouse#39055] [Q&A] [answered]
i see in [document](https://clickhouse.com/docs/en/engines/database-engines/atomic#drop-detach-table) that 

> Table will be actually removed when it's not in use.

my question is: when I execute "drop table tableA on cluster `{cluster}` no delay", will this table be dropped successfully even if it has continually insert (it's the source table of a Materialized View）

**Comment by tavplubix** [comment, marked as answer]
Yes, it will be dropped successfully.
> Table will be actually removed when it's not in use.

It means that data is actually removed asynchronously in background when table is not in use. `NO DELAY` modifier (it's deprecated synonym of `SYNC` modifier) means "ignore `database_atomic_delay_before_drop_table_sec` and and wait for data to be actually removed". So in this case DROP query will mark table as dropped immediately (so table will not be available for new queries) and then will wait for already running queries to finish so data can be removed. Eventually DROP will finish successfully.

P.S. It's highly not recommended to run synchronous queries ON CLUSTER. It's because all ON CLUSTER queries are executed in single thread and synchronous query may make this thread (and the whole distributed DDL queue) stuck for a while just waiting for something.
